components/Sidebar/panels/Playback/Playback.tsx

```diff
@@ -0,0 +1,148 @@
+"use client";
+
+import { useRouter } from "next/navigation";
+import { useState, useTransition } from "react";
+
+import getCurrentTrackUri from "@/lib/playback/gettrackrui";
+
+import play from "../../../../lib/playback/play";
+import next from "../../../../lib/playback/skipnext";
+import prev from "../../../../lib/playback/skipprev";
+import BasePanel from "../BasePanel";
+
+let currentIsTrackPlaying = false;
+
+const Playback = ({
+  isTrackPlaying,
+  uri,
+}: {
+  isTrackPlaying: boolean;
+  uri: string;
+}) => {
+  const router = useRouter();
+  const [isPending, startTransition] = useTransition();
+  const [isFetching, setIsFetching] = useState(false);
+  const [isPlaying, setIsPlaying] = useState(false); // new state variable for player state
+
+  const handlePrevClick = async (e: React.MouseEvent) => {
+    e.preventDefault();
+    setIsFetching(true);
+    await prev(uri, isTrackPlaying, true);
+    setIsFetching(false);
+    startTransition(() => {
+      router.refresh();
+    });
+  };
+  const handleNextClick = async (e: React.MouseEvent) => {
+    e.preventDefault();
+    setIsFetching(true);
+    await next(uri, isTrackPlaying, true);
+    setIsFetching(false);
+    startTransition(() => {
+      router.refresh();
+    });
+  };
+
+  const handleClick = async (e: React.MouseEvent) => {
+    e.preventDefault();
+    setIsFetching(true);
+    // TODO unsafe access on any value
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const tmp_uri = await getCurrentTrackUri();
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument, @typescript-eslint/no-unsafe-member-access
+    currentIsTrackPlaying = await play(tmp_uri.item.uri, currentIsTrackPlaying);
```

Comment:
Clicking play when a song is paused resets the song. You shouldn't pass in the URI if you just want to resume the current song, only if youre starting a new song